app/services/unpublish_handler_service.rb

```diff
@@ -0,0 +1,89 @@
+class UnpublishHandlerService
+  UNPUBLISH_LOG_PATH = "#{Rails.root}/log/unpublish_message.log".freeze
+
+  def initialize(content_id)
+    @content_id = content_id
+  end
+
+  def self.call(*args)
+    new(*args).call
+  end
+
+  def call
+    if email_params.empty?
```

Comment:
email_params is also called in line 16 and could be memoized

Reply:
You could also make UnpublishEmailBuilder deal with an empty array

Reply:
Yep i could memoize email_params, however because I memoized `subscriber_lists` I assumed it would be okay to leave.

However i'll explore making `UnpublishEmailBuilder` handle an empty array instead, which will make this problem obsolete and probably make the code look nicer!